misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -15,24 +15,283 @@
     }
 
-    static final String NEWLINE_SYMBOL = System.getProperty("line.separator");
+    /**
+     * Checks whether the module is a build project.
+     *
+     * @param module Current module
+     * @return whether module is a build project
+     */
+    public static boolean isBuildProject(Optional<Module> module) {
+        return module.isPresent() && module.get().project().kind() != ProjectKind.SINGLE_FILE_PROJECT;
+    }
+
+    /**
+     * Retrieves the default boolean value for the format option in the given format section.
+     *
+     * @param section the format section
+     * @param key     the format option
+     * @return the default boolean value
+     */
+    public static boolean getDefaultBoolean(FormatSection section, String key) {
+        return Boolean.parseBoolean(getDefaultString(section, key));
+    }
+
+    /**
+     * Retrieves the default integer value for the format option in the given format section.
+     *
+     * @param section the format section
+     * @param key     the format option
+     * @return the default integer value
+     */
+    public static int getDefaultInt(FormatSection section, String key) {
+        return Integer.parseInt(getDefaultString(section, key));
+    }
+
+    /**
+     * Retrieves the default string value for the format option in the given format section.
+     *
+     * @param section the format section
+     * @param key     the format option
+     * @return the default string value
+     */
+    public static String getDefaultString(FormatSection section, String key) {
+        return DEFAULTS.getString(section.getStringValue() + "." + key);
+    }
+
+    /**
+     * Loads the format section in the Ballerina.toml .
+     *
+     * @param manifest the package manifest
+     * @return the format section
+     */
+    public static Object loadFormatSection(PackageManifest manifest) {
+        return manifest.getValue(FORMAT);
+    }
+
+    /**
+     * Retrieves the formatting file path.
+     *
+     * @param formatSection the format section loaded from Ballerina.toml
+     * @param root          the root path of the project
+     * @return the formatting file path
+     */
+    public static Optional<String> getFormattingFilePath(Object formatSection, String root) {
+        if (formatSection != null) {
+            Object path = ((Map<String, Object>) formatSection).get(FORMAT_FILE_FIELD);
+            if (path != null) {
+                String str = path.toString();
+                if (str.endsWith(FORMAT_OPTION_FILE_EXT)) {
+                    return Optional.of(str);
+                }
+                return Optional.of(Path.of(str, DEFAULT_FORMAT_OPTION_FILE).toString());
+            }
+        }
+
+        Path defaultFile = Path.of(root, DEFAULT_FORMAT_OPTION_FILE);
+        return Files.exists(defaultFile) ? Optional.of(defaultFile.toString()) : Optional.empty();
+    }
+
```

Comment:
Do we have tests for the prioritization in-line with https://github.com/ballerina-platform/ballerina-lang/issues/41280#issuecomment-1769311138?

Reply:
> * If the format file path is specified in `Ballerina.toml`, then format package sources based on the config in that file.
>   
>   * Path has to be either of:
>     
>     * Absolute file path,
>     * Relative file path from the package root,
>     * HTTP or HTTPS URL

We do have remote (relative path. from project root) and local file tests.
https://github.com/poorna2152/ballerina-lang/blob/formatter/misc/formatter/modules/formatter-core/src/test/resources/configurations/remote/source/Ballerina.toml


> * If the file path is not in `Ballerina.toml`, look for a `Format.bal` in the current package root directory.
> * If no `Format.bal` exists in the package root, then use the default format configs.

We have tests where path is not in Format.toml and we resolve Format.toml file in the project directory. 
https://github.com/poorna2152/ballerina-lang/tree/formatter/misc/formatter/modules/formatter-cli/src/test/resources/configurations/module/source/project